sys/hash_string/hash_string.c

```diff
@@ -24,9 +24,8 @@
 unsigned long hash_string(unsigned char *str)
 {
     unsigned long hash = 5381;
-    int c;
 
-    while ((c = *str++)) {
+    for (char c; (c = *str) != 0; ++str) {
```

Comment:
As stated before: this change is neither related to the unittests nor is it presenting any benefits apart from cosmetic ones based on personal taste. I would prefer to have the changes to this file removed from this PR.